Problem2/src/problem2/Problem2.java

```diff
@@ -9,8 +9,44 @@
     /**
      * @param args the command line arguments
      */
-    public static void main(String[] args) {
-        // TODO code application logic here
+    public static void main(String[] args)
+    {
+        Scanner scan = new Scanner(System.in); //Declare/set variables
+        System.out.println("Enter the number you would like to check up to.");
+        int size = scan.nextInt();
+        boolean[] primes = new boolean[size+1];
+        for (int i = 0; i < primes.length; i++)
+        {
+            primes[i] = true;
+        }
+        
+        //Loop to set non-primes to false
+        for (int i = 2; i <= Math.sqrt(size); i++) //Runs through every number up to the squre root of size
+        {
+            for(int j = i+1; j <= size; j++) //Checks if every number larger than i is a multiple of i
+            {
+                if (j%i == 0)
```

Comment:
You get the right answer, but your algorithm is not quite right.  You’re supposed to skip the number if it is already deemed to be not prime.  Also, if the number is prime, you’re supposed to count by multiples of the prime number to update the array.